### ISSUE TITLE
Generic (non-analytic) model support

### ISSUE THREAD

**Issue by dlakaplan** [sidmohite/kNe-inference#3]

**Comment by sidmohite** [comment]
On thinking about this further as well, wouldn't any model for the absolute mag evolution depend on some parameter after all. I mean I get the aspect of a numerical model where there won't be any direct dependence on a parameter but it could indirectly depend on things like ejecta mass, opening angle etc. My point being that eventually there will always be a map F (param1,param2,.... )  =  M(t). I guess the inference then in that case could be over these parameters then. Currently what I have is agnostic about the functional form:
`M_g = np.array([self.lc_model_g(*params, t_0=t0, t=t_j)])`

**Comment by dlakaplan** [comment]
Yes, definitely.  I just didn't know if your code could handle models say without analytic models/derivatives.  Or perhaps discrete variables.  But as long as you built it generically that's good.
